#! /bin/sh /usr/share/dpatch/dpatch-run

## 05_segfault_at_nonexisting_device.dpatch
## by Ludwig Nussel <ludwig.nussel@suse.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix segfault when trying to open a non existing device

@DPATCH@

--- cryptsetup-1.0.5/lib/setup.c
+++ cryptsetup-1.0.5/lib/setup.c
@@ -606,7 +606,7 @@
 
 static int __crypt_luks_open(int arg, struct setup_backend *backend, struct crypt_options *options)
 {
-	struct luks_masterkey *mk;
+	struct luks_masterkey *mk=NULL;
 	struct luks_phdr hdr;
 	char *password; int passwordLen;
 	struct device_infos infos;
